pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
We should figure out if there is some reason for this sleep, it seems like something that was put there for a reason. Perhaps the original author was worried it would be restarted by kubernetes before the node was rebooted and screw up the node labels or some other subtle problem. If it isn't needed, however, I totally agree that it shouldn't even exist. (@euank may know more about this?) 

The reason I was pointing out the `stop` channel not being closed is that this will regress on #93. The goroutines started by `process` need to be stopped before the node is told to reboot, or else they will panic.